carbonmark/components/pages/Portfolio/ConnectedWallet/index.tsx

```diff
@@ -0,0 +1,72 @@
+import { concatAddress, isTorusProvider, useWeb3 } from "@klimadao/lib/utils";
+import { Trans } from "@lingui/macro";
+import { Card } from "components/Card";
+import { Text } from "components/Text";
+import { FC, useEffect, useState } from "react";
+import { ViewWalletButton } from "components/ViewWalletButton";
+import { CopyAddressButton } from "components/CopyAddressButton";
+import { urls } from "@klimadao/lib/constants";
+import { useCopyToClipboard } from "@klimadao/lib/hooks";
+import { cx } from "@emotion/css";
+import * as styles from "./styles";
+import { ButtonPrimary } from "@klimadao/lib/components";
+import ContentCopyIcon from "@mui/icons-material/ContentCopy";
+import CheckIcon from "@mui/icons-material/Check";
+import LaunchIcon from "@mui/icons-material/Launch";
+
+type Props = {
+  address: string | undefined;
```

Comment:
We should make this required and the parent component should not show the card at all if the user is not connected / there is no address
see figma (no card shown here https://www.figma.com/file/I5hXFeW1DKjV7agHq7T4UZ/Portfolio?type=design&node-id=8-1256&mode=design&t=6glrZlg3jR00RgdT-0)

```ts
{props.address && <ConnectedWallet address={props.address} />}
```